client/gutenberg/extensions/map/component.js

```diff
@@ -0,0 +1,328 @@
+import InfoWindow from './info-window/';
+import { mapboxMapFormatter } from './mapbox-map-formatter/';
+
+export class Map extends Component {
+	// Lifecycle
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			map: null,
+			fit_to_bounds: false,
+			loaded: false,
+			mapboxgl: null,
+		};
+
+		// Refs
+		this.mapRef = createRef();
+
+		// Debouncers
+		this.debouncedSizeMap = debounce( this.sizeMap, 250 );
+	}
+	render() {
+		const { points, admin, children, marker_color } = this.props;
+		const { map, activeMarker, mapboxgl } = this.state;
+		const { onMarkerClick, deleteActiveMarker, updateActiveMarker } = this;
+		const currentPoint = get( activeMarker, 'props.point' ) || {};
+		const { title, caption } = currentPoint;
+		let addPoint = null;
+		Children.map( children, element => {
+			if ( element && 'AddPoint' === element.type.name ) {
+				addPoint = element;
+			}
+		} );
+		const mapMarkers =
+			map &&
+			mapboxgl &&
+			points.map( ( point, index ) => {
+				return (
+					<MapMarker
+						key={ index }
+						point={ point }
+						index={ index }
+						map={ map }
+						mapboxgl={ mapboxgl }
+						marker_color={ marker_color }
+						onClick={ onMarkerClick }
+					/>
+				);
+			} );
+		const infoWindow = mapboxgl && (
+			<InfoWindow
+				activeMarker={ activeMarker }
+				map={ map }
+				mapboxgl={ mapboxgl }
+				unsetActiveMarker={ () => this.setState( { activeMarker: null } ) }
+			>
+				{ activeMarker &&
+					admin && (
+						<Fragment>
+							<TextControl
+								label={ __( 'Marker Title' ) }
+								value={ title }
+								onChange={ value => updateActiveMarker( { title: value } ) }
+							/>
+							<TextareaControl
+								className="wp-block-jetpack-map__marker-caption"
+								label={ __( 'Marker Caption' ) }
+								value={ caption }
+								rows="2"
+								tag="textarea"
+								onChange={ value => updateActiveMarker( { caption: value } ) }
+							/>
+							<Button onClick={ deleteActiveMarker } className="wp-block-jetpack-map__delete-btn">
+								<Dashicon icon="trash" size="15" /> { __( 'Delete Marker' ) }
+							</Button>
+						</Fragment>
+					) }
+
+				{ activeMarker &&
+					! admin && (
+						<Fragment>
+							<h3>{ title }</h3>
+							<p>{ caption }</p>
+						</Fragment>
+					) }
+			</InfoWindow>
+		);
+		return (
+			<Fragment>
+				<div className="wp-block-jetpack-map__gm-container" ref={ this.mapRef }>
+					{ mapMarkers }
+				</div>
+				{ infoWindow }
+				{ addPoint }
+			</Fragment>
+		);
+	}
+	componentDidMount() {
+		const { api_key } = this.props;
+		if ( api_key ) {
+			this.loadMapLibraries();
+		}
+	}
+	componentWillUnmount() {
+		this.debouncedSizeMap.cancel();
+	}
+	componentDidUpdate( prevProps ) {
+		const { api_key, children, points, map_style, map_details } = this.props;
+		const { map } = this.state;
+		if ( api_key && api_key.length > 0 && api_key !== prevProps.api_key ) {
+			window.mapboxgl = null;
```

Comment:
What's this for? Maybe left over?

Reply:
Absolutely correct - I'll remove.

Reply:
Removed: https://github.com/Automattic/wp-calypso/pull/28498/commits/fb1156c97440a5276fbe1a727365709d21794bf0